plugin/src/main/java/com/denizenscript/denizen/events/block/CrafterCraftsScriptEvent.java

```diff
@@ -0,0 +1,86 @@
+package com.denizenscript.denizen.events.block;
+
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import org.bukkit.Keyed;
+import org.bukkit.Material;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.CrafterCraftEvent;
+import org.bukkit.inventory.ItemStack;
+
+public class CrafterCraftsScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // crafter crafts <item>
+    //
+    // @Group Block
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers when a crafter crafts an item.
+    //
+    // @Context
+    // <context.location> returns the LocationTag of the crafter.
+    // <context.item> returns the ItemTag of the item being crafted.
+    // <context.recipe_id> returns the ElementTag of the recipe ID formed.
+    //
+    // @Determine
+    // ItemTag to set the item being crafted.
+    //
+    // -->
+
+    public CrafterCraftsScriptEvent() {
+        registerCouldMatcher("crafter crafts <item>");
+        this.<CrafterCraftsScriptEvent, ItemTag>registerDetermination("item", ItemTag.class, (evt, context, result) -> {
+            event.setResult(result.getItemStack());
```

Comment:
What happens to the ingredients when you `setResult`? Are they still consumed, or not? Is it different with `air` (when it's cancelled)? Make sure the answers to all questions like this make sense, and document anything that isn't obvious.